### ISSUE TITLE
"The specified key does not exist." - crashes SQS polling

### ISSUE THREAD

**Issue by JoeOrtiz** [fluent/fluent-plugin-s3#167]
It appears like if the s3 key is not found in the bucket, SQS polling just halts, anyway to let it continue?

```
Oct 14 00:35:28 2016-10-14 00:35:28 +0000 [warn]: The specified key does not exist.
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.6.6/lib/seahorse/client/plugins/raise_response_errors.rb:15:in `call'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.6.6/lib/aws-sdk-core/plugins/s3_sse_cpk.rb:19:in `call'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.6.6/lib/aws-sdk-core/plugins/s3_dualstack.rb:23:in `call'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.6.6/lib/aws-sdk-core/plugins/s3_accelerate.rb:33:in `call'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.6.6/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.6.6/lib/seahorse/client/plugins/response_target.rb:21:in `call'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.6.6/lib/seahorse/client/request.rb:70:in `send_request'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.6.6/lib/seahorse/client/base.rb:207:in `block (2 levels) in define_operation_methods'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/request.rb:24:in `call'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/operations.rb:41:in `call'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/operation_methods.rb:19:in `block in add_operation'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-s3-0.7.1/lib/fluent/plugin/in_s3.rb:218:in `process'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-s3-0.7.1/lib/fluent/plugin/in_s3.rb:149:in `block in run'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:413:in `block in yield_messages'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:412:in `each'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:412:in `yield_messages'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:405:in `block in process_messages'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:404:in `catch'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:404:in `process_messages'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:336:in `block (2 levels) in poll'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:331:in `loop'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:331:in `block in poll'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:330:in `catch'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.6/lib/aws-sdk-resources/services/sqs/queue_poller.rb:330:in `poll'
Oct 14 00:35:28 /usr/lib/ruby/gems/2.3.0/gems/fluent-plugin-s3-0.7.1/lib/fluent/plugin/in_s3.rb:144:in `run'

```

**Comment by repeatedly** [comment]
@okkez Could you check this?

**Comment by okkez** [comment]
Change `@running` to `false` at  https://github.com/fluent/fluent-plugin-s3/blob/v0.7.1/lib/fluent/plugin/in_s3.rb#L152
This stops polling SQS queue.

If `@running` is not changed here, polling will not stop.
In this case, errored message is not removed and this plugin will retry processing errored messages.

What should I do?

**Comment by repeatedly** [comment]
Fixed via #170
